gluten-core/src/main/scala/io/glutenproject/execution/TakeOrderedAndProjectExecTransformer.scala

```diff
@@ -68,32 +67,40 @@ case class TakeOrderedAndProjectExecTransformer(limit: Int,
     if (childRDDPartsNum == 0) {
       sparkContext.parallelize(Seq.empty, 1)
     } else {
+      val orderingSatisfies = SortOrder.orderingSatisfies(child.outputOrdering, sortOrder)
+      def withLocalSort(sparkPlan: SparkPlan): SparkPlan = {
+        if (orderingSatisfies) {
+          sparkPlan
+        } else {
+          SortExecTransformer(sortOrder, false, sparkPlan)
```

Comment:
Just out of curiosity (lack of some backgrounds), does the code mean we are not using optimized algorithms for the top-k calculation here? I noticed that vanilla Spark calls Guava's top-k which is ideally O(n).

@lgbo-ustc

Reply:
It seems velox support topN. I'm not sure why we did not support transform TakeOrderedAndProjectExec to TopN before. This pr itself just removes a local sort based on previous code.